packages/maskbook/src/utils/hooks/useImageFailover.ts

```diff
@@ -0,0 +1,26 @@
+import { useAsync } from 'react-use'
+
+interface ImgCdnPair {
+    img: string
+    cdn: string
+}
+
+export function useImageFailover(imgCdnPairs: ImgCdnPair[]) {
+    return useAsync(() => {
+        return new Promise((resolve) => {
+            const image = new Image()
+            let imgCdnPair: ImgCdnPair
+            image.addEventListener('error', () => {
+                if (imgCdnPairs.length === 0) resolve('')
+                imgCdnPair = imgCdnPairs.shift()!
```

Comment:
hold on, this is mutating the original array.

Reply:
So I need to declare a private variable.

Reply:
so you need to avoid mutating the array. change the signature to readonly

Reply:
done